streams/src/main/java/org/apache/kafka/streams/kstream/internals/WrappingNullableSerde.java

```diff
@@ -60,9 +60,12 @@ public void close() {
     }
 
     public void setIfUnset(final Serde<InnerK> defaultKeySerde, final Serde<InnerV> defaultValueSerde) {
-        Objects.requireNonNull(defaultKeySerde);
-        Objects.requireNonNull(defaultValueSerde);
-        serializer.setIfUnset(defaultKeySerde.serializer(), defaultValueSerde.serializer());
-        deserializer.setIfUnset(defaultKeySerde.deserializer(), defaultValueSerde.deserializer());
+        if (defaultKeySerde != null && defaultValueSerde != null) {
```

Comment:
Why is this a single condition? Should we not handle keySerde and valueSerde independently? -- Atm, if one is null, but the other is not null, we would execute the `else` and set _both_ to `null`; is this intentional?

Also, following the comment from above, should we pass in the context into this method?

Reply:
The context issue is same as above - we need to be able to handle two processor types. The reason I left these lumped together is because in the call to `setIfUnset` we pass in the `serializer`/`deserializer`, so if we let there be a null in either we'd get a NPE at the method call. I tried passing in just the `default` for both but ran into issues wtih the generics. I can refactor to allow one to be null